src/ilasm/CMakeLists.txt

```diff
@@ -60,12 +60,12 @@ if(CLR_CMAKE_PLATFORM_UNIX)
     palrt
   )
 
-  # FreeBSD implements dlopen in libc
-  if(NOT CMAKE_SYSTEM_NAME STREQUAL FreeBSD)
+  # FreeBSD and NetBSD implement dlopen(3) in libc
+  if(NOT CMAKE_SYSTEM_NAME STREQUAL FreeBSD AND NOT CMAKE_SYSTEM_NAME STREQUAL NetBSD)
     target_link_libraries(ilasm
       dl
```

Comment:
The same thing is in `src\coreclr\hosts\unixcoreconsole\CMakeLists.txt`, `src\coreclr\hosts\unixcorerun\CMakeLists.txt` and `src\ildasm\exe\CMakeLists.txt`, so it seems it would make sense to change it in this fix as well.